python/selfie-lib/selfie_lib/SourceFile.py

```diff
@@ -0,0 +1,155 @@
+from Literals import Language
+from EscapeLeadingWhitespace import EscapeLeadingWhitespace
+
+
+class SourceFile:
+    TRIPLE_QUOTE = '"""'
+
+    def __init__(self, filename, content):
+        self.unix_newlines = "\r" not in content
+        self.content_slice = content.replace("\r\n", "\n")
+        self.language = Language.from_filename(filename)
+        self.escape_leading_whitespace = EscapeLeadingWhitespace.appropriate_for(
+            self.content_slice
+        )
+
+    @property
+    def as_string(self):
+        return (
+            self.content_slice
+            if self.unix_newlines
+            else self.content_slice.replace("\n", "\r\n")
+        )
+
+    class ToBeLiteral:
+        def __init__(self, dot_fun_open_paren, function_call_plus_arg, arg):
```

Comment:
Types will be super helpful throughout this file, but especially here. I think some of these are `Slice` and some are `String`, very hard to keep track of which is which without static types.